### ISSUE TITLE
Determine whether user possesses game command line script

### ISSUE THREAD

**Issue by tatsujb** [ValveSoftware/steam-for-linux#6218]
#### Your system information

* Steam client version (build number or date): Feb 18 2019
* Distribution (e.g. Ubuntu): Ubuntu
* Opted into Steam client beta?: [Yes]
* Have you checked for system updates?: [Yes]

#### Please describe your issue in as much detail as possible:

I'm writing a script here : https://github.com/tatsujb/installFAFscript

It installs the steam game 9420 (supreme commander forged alliance) and configures it to work in tandem with it's unofficial (but crushingly prevalent) multiplayer client : (FAF)

I feel I may eventually get traffic from new users in which case the script looses sense if the user hasn't purchased 9420 on steam first + made his steam library public https://faforever.com/2019/01/copyright-issues.

I'd like to be able to detect this use-case in order to open up a web-browser to the steam store page for Forged Alliance or something of the nature and kill the script.

I can always look at `~/.steam/steam/steamapps/libraryfolders.vdf` but this only applies for users who have steam installed and logged into pre-emptively. 

other wise I'll have installed and run a bunch of things before I can tell the user "oh, sorry, got to uninstall and re-run when you own Forged Alliance"

of course if the user has not made his steam library public, my script would warn the user of this and close as well as FAF does not allow the user to log in if you haven't made your steam library public this and do not possess FA on steam.

any ideas?

**Comment by kisak-valve** [comment]
Sounds like something to try logging in with steamcmd and licenses_for_app 9420.

Example anonymous example for TF2: `./steamcmd.sh +login anonymous +licenses_for_app 440 +quit`

**Comment by tatsujb** [comment]
no? how would that make a valid if case for whether the user possesses the game or not?

**Comment by tatsujb** [comment]
@kisak-valve that command only works for games without DRM.

**Comment by kisak-valve** [comment]
I'd expect `+login anonymous` to only work with the [anonymous dedicated server comp](https://steamdb.info/sub/17906/apps/), you'd need to prompt the user for their login credentials to do a command line check of the subs a user has from a script. The point of the example was to show a simple working example of `+licenses_for_app`.

**Comment by tatsujb** [comment]
@kisak-valve ah my bad. that works. 

Do you think it's possible I get pointers for my script (https://github.com/tatsujb/installFAFscript) on a new issue here?  : I'm pretty sure I'm missing out on good insight the people who wrote these tools might have, such as a technique that would maintain an open session so as to prompt the user for steam guard only once throughout the use of my script instead of the current 3 (going on 4 with the added check of steam library) ideally 0 if the user has already typed it in once before for that computer and his steam is currently set to remember password.

in such a use-case everytime the user starts steam he is simply logged in without being prompted for steam guard. 

surely steamCMD and steam (run from terminal) prompting for steam guard code in this use-case is not meant to be intended behavior?

is there at least a way to tell steam to ignore the `+login ` command and matching arguments if upon being called steam detects an ongoing session already open?

this would help streamline my script a whole lot.

**Comment by Chiitoo** [comment]
I only skimmed the script, so I don't have too much to say about it at this time, but when I see 'eval', I read 'evil'.  :]

You may have already taken into consideration the security issues of using 'eval', but just in case, see for example: http://mywiki.wooledge.org/BashFAQ/048

**Comment by tatsujb** [comment]
@Chiitoo I know. I'm knew to this and I haven't yet found another functioning solution. I'm looking for contributors and maintainers, would you be interested?

**Comment by Chiitoo** [comment]
I'm definitely no expert myself, and sadly am out of time most of the time... but I will definitely keep the thing in mind, should I have more of it for a more thorough look.  :]

If you haven't yet been there, I would definitely suggest hanging out in #bash at freenode.  Even just idling and reading things others say there can give a lot of good pointers, and I can't think of a better place to ask questions regarding the subject.  ^^

**Comment by tatsujb** [comment]
thanks, will do! I do however need steam staff help on things such as setting a games launch arguments and enabling proton as well as better login support all via command-line to make my script better. 

these are the main issues that downgrade my script into something where the user still has to do a lot of things manually and intervene all throughout. And this has no good reason to be.
